int8 device_id
int8 clamp_num
float64 power